mongo/collection.go

```diff
@@ -442,14 +442,23 @@ func (coll *Collection) InsertOne(ctx context.Context, document interface{},
 // The opts parameter can be used to specify options for the operation (see the options.InsertManyOptions documentation.)
 //
 // For more information about the command, see https://www.mongodb.com/docs/manual/reference/command/insert/.
-func (coll *Collection) InsertMany(ctx context.Context, documents []interface{},
+func (coll *Collection) InsertMany(ctx context.Context, documents interface{},
 	opts ...*options.InsertManyOptions) (*InsertManyResult, error) {
 
-	if len(documents) == 0 {
+	dv := reflect.ValueOf(documents)
```

Comment:
The `_switch` implementation, while being more performant, does not improve the API because users would still be forced to convert from `[]Document` to `[]interface{}`.

Reply:
Explicitly accepting `[]interface{}` (what we're already doing)
```
func (coll *Collection) InsertMany(ctx context.Context, documents []interface{},
	opts ...*options.InsertManyOptions) (*InsertManyResult, error) {
```

is better than accepting `interface{}` with the implicit requirement that `documents` must be `[]interface{}`
```
func (coll *Collection) InsertMany(ctx context.Context, documents interface{},
	opts ...*options.InsertManyOptions) (*InsertManyResult, error) {
```

Reply:
The solution I gave is just a primer, you  would still need to make it conform generically using the `bson` package. I think it's acceptable to merge this PR. I've opened another ticket to remove the reflection logic and optimize later down the line: GODRIVER-3026